lambdas/nbconvert/index.py

```diff
@@ -1,34 +1,105 @@
+from enum import Enum
+import json
+import os
 import urllib.request
 
+from jsonschema import Draft4Validator, ValidationError
 from nbconvert import HTMLExporter
 import nbformat
 import pandas as pd
-from urllib.parse import urlparse
+from urllib.parse import urlparse, unquote
+
+WEB_ORIGIN = os.environ['WEB_ORIGIN']
+
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'url': {
+            'type': 'string'
+        },
+        'output': {
+            'enum': ['html', 'json']
+        },
+        'input': {
+            'enum': ['ipynb', 'parquet']
+        },
+        'nbconvert_template': {
+            'enum': ['full', 'basic']
+        }
+    },
+    'required': ['url'],
+    'additionalProperties': False
+}
+
+Draft4Validator.check_schema(SCHEMA)
+VALIDATOR = Draft4Validator(SCHEMA)
 
-html_exporter = HTMLExporter()
-html_exporter.template_file = 'full'
 
 def lambda_handler(event, context):
-    url = event['queryStringParameters'].get('url')
-    if not url:
+    params = event['queryStringParameters'] or {}
+    headers = event['headers'] or {}
+
+    try:
+        VALIDATOR.validate(params)
+    except ValidationError as ex:
         return {
-            "body": "Missing 'url'",
+            "body": ex.message,
             "statusCode": 400
         }
 
-    parse = urlparse(url)
-    if parse.path.endswith('.parquet'):
-        df = pd.read_parquet(url)
-        body = df._repr_html_()
-    else:
+    url = params['url']
+    input_type = params.get('input')
+
+    # TODO(dima): Make it required.
+    if input_type is None:
+        parsed_url = urlparse(url)
+        path = unquote(parsed_url.path)
+        if path.endswith('.parquet'):
+            input_type = 'parquet'
+        else:
+            input_type = 'ipynb'
+
+
+    if input_type == 'ipynb':
+        html_exporter = HTMLExporter()
+        html_exporter.template_file = params.get('nbconvert_template', 'full')
+
         response = urllib.request.urlopen(url).read().decode()
         notebook = nbformat.reads(response, 4)
-        body, resources = html_exporter.from_notebook_node(notebook)
+        result, _ = html_exporter.from_notebook_node(notebook)
```

Comment:
This is just a variable. Do you mean, just return JSON and drop plain HTML support?

Reply:
i mean the response body should always be a JSON string with an `html` key. then, for every preview type, the client can *always* look in `body.html`, and optionally `body.meta`... instead of doing different things for each file type.